src/teacher.py

```diff
@@ -77,23 +77,30 @@ def _answer_Demonstration_query(
 
 
 @dataclass
-class TeacherInteractionMonitor(utils.Monitor):
-    """Wraps the interaction between agent and teacher to include generating
-    and answering queries."""
+class TeacherInteractionMonitor(utils.VideoMonitor):
+    """A monitor that renders each state and action encountered and queries and
```

Comment:
nit: grammar

Reply:
Let's rewrite this comment. The important thing about this Monitor isn't that it renders, it's about the querying. Maybe: "A monitor that wraps interaction with an environment to include generating and answering queries. Optionally, if CFG.make_interaction_videos is True, also renders every state and action encountered."